server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -79,6 +79,22 @@ public void testFieldDisabled() throws Exception {
         assertNotNull(doc.rootDoc().getField(IdFieldMapper.NAME));
     }
 
+    public void testDotsWithFieldDisabled() throws IOException {
+        DocumentMapperParser mapperParser = createIndex("test").mapperService().documentMapperParser();
+        String mapping = Strings.toString(jsonBuilder().startObject().startObject("type").startObject("properties")
+            .startObject("foo").field("enabled", false).endObject()
+            .endObject().endObject().endObject());
+        DocumentMapper mapper = mapperParser.parse("type", new CompressedXContent(mapping));
+        BytesReference bytes = BytesReference.bytes(jsonBuilder()
```

Comment:
It would be great to test some other types as fields as well, such as array fields, since they go through a different parsing code path than simple values.